File: crossecore-java/src/java_/TextElementBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class TextElementBase 
extends ASTNodeImpl implements TextElement
{
	protected static final java.lang.String TEXT_EDEFAULT = "";
	private java.lang.String _text = TEXT_EDEFAULT;
	public java.lang.String getText()
	{
		return _text;
	}
	public void setText(java.lang.String value){
		
		java.lang.String oldValue = _text;
		_text = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.TEXTELEMENT_TEXT, oldValue, value));
		
	}
	
	
	

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.TEXTELEMENT;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.TEXTELEMENT_TEXT:
				return getText();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.TEXTELEMENT_TEXT:
				setText((java.lang.String) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.TEXTELEMENT_TEXT:
				return TEXT_EDEFAULT == null ? _text != null : !TEXT_EDEFAULT.equals(_text);
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/TextElementBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class TextElementBase 
	:ASTNodeImpl, TextElement
	{
		private string _text = "";
		public virtual string text
		{
		get { 
			return _text;
		}
		set { _text = value; }
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.TEXTELEMENT;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.TEXTELEMENT_TEXT:
					return text;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.TEXTELEMENT_TEXT:
					text = (string) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.TEXTELEMENT_TEXT:
					return TEXT_EDEFAULT == null ? _text != null : !TEXT_EDEFAULT.equals(_text);
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
